smart-contracts/test/Lock/destroyLock.js

```diff
@@ -0,0 +1,91 @@
+const Units = require('ethereumjs-units')
+const Web3Utils = require('web3-utils')
+const BigNumber = require('bignumber.js')
+
+const deployLocks = require('../helpers/deployLocks')
+const shouldFail = require('../helpers/shouldFail')
+const Unlock = artifacts.require('../Unlock.sol')
+
+let unlock, locks
+
+contract('Lock', accounts => {
+  let lock
+
+  before(async () => {
+    unlock = await Unlock.deployed()
+    locks = await deployLocks(unlock)
+    lock = locks['FIRST']
+  })
+
+  describe('destroyLock', () => {
+    it('should fail if called by the wrong account', async () => {
+      await shouldFail(lock.destroyLock({ from: accounts[1] }), '')
+    })
+
+    describe('when called by the owner', () => {
+      let initialLockBalance, initialOwnerBalance, txObj, event
+
+      before(async () => {
+        await lock.purchaseFor(accounts[1], Web3Utils.toHex('Julien'), {
+          value: Units.convert('0.01', 'eth', 'wei')
+        })
+        assert.equal(await lock.getHasValidKey(accounts[1]), true) // pre-req
+
+        initialLockBalance = new BigNumber(
+          await web3.eth.getBalance(lock.address)
+        )
+        initialOwnerBalance = new BigNumber(
+          await web3.eth.getBalance(accounts[0])
+        )
+        await lock.disableLock() // We can't destroy a lock without first disabling it
+        txObj = await lock.destroyLock({ from: accounts[0] })
+        event = txObj.logs[0]
+      })
+
+      it('owner should have received funds from the contract', async () => {
+        const finalOwnerBalance = new BigNumber(
+          await web3.eth.getBalance(accounts[0])
+        )
+        assert(finalOwnerBalance.gt(initialOwnerBalance))
+      })
+
+      it('contract should no longer have any funds', async () => {
+        const finalLockBalance = new BigNumber(
+          await web3.eth.getBalance(lock.address)
+        )
+        assert.equal(finalLockBalance.toFixed(), 0)
+      })
+
+      it('should trigger the Destroy event', () => {
+        assert.equal(event.event, 'Destroy')
+        assert(event.args.balance.eq(initialLockBalance))
+        assert.equal(event.args.owner, accounts[0])
+      })
+
+      it('previously valid key is no longer valid', async () => {
+        assert.equal(await lock.getHasValidKey(accounts[1]), false)
```

Comment:
With solidity-coverage coming, this will break the build.  Read only calls must use `.call` or they fail in that environment.

Reply:
got it.